<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel = "stylesheet" href = "https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
    <!-- Select2 CSS -->
<link href="https://cdn.jsdelivr.net/npm/select2@4.1.0-beta.1/dist/css/select2.min.css" rel="stylesheet" />
<script src = "https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"> </script>
    <!-- Select2 JS -->
<script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-beta.1/dist/js/select2.min.js"></script>
 <script src = "../js/menu.js"></script> <!-- jp - import menu.js on every page -->
<script src = "../js/GeneExp.js"></script> <!-- jp import page-specific js file -->
<script src = "https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
<script src = "https://cdnjs.cloudflare.com/ajax/libs/bootstrap-multiselect/0.9.15/js/bootstrap-multiselect.min.js"></script>
<link rel = "stylesheet" href = "https://cdnjs.cloudflare.com/ajax/libs/bootstrap-multiselect/0.9.15/css/bootstrap-multiselect.css"/>
<link rel="stylesheet" href="{{url_for('static', filename='css/style.css')}}">


<script src="https://cdn.datatables.net/1.10.24/js/jquery.dataTables.min.js"></script>
     <link rel="stylesheet" href="https://cdn.datatables.net/1.10.24/css/jquery.dataTables.min.css" />
     <style>
      .multiselect-container {
        width: 100% !important;
    }
     .select2-selection__rendered {
    line-height: 31px !important;
}
.select2-container .select2-selection--single {
    height: 35px !important;
}
.select2-selection__arrow {
    height: 34px !important;
}
        div.dataTables_wrapper {
        width: 800px;
        margin: 0 auto;
    }

#sidebar {
    min-width: 380px;
    max-width: 380px;
}

p {
text-align: justify;
text-justify: inter-word;
}

     </style>
</head>
<body>
<div class="wrapper">
            <!-- Sidebar Holder -->
            <nav id="sidebar">
                <div class="sidebar-header">
                    <h3>Gene Expression Features</h3>
                </div>

<form method="post" id="framework_form">
<!--<form method="post" id="framework_form" action="/featuresBoot">-->
<div class="container">
<h4>Gene Expression </h4>
<select name="comp_select"  id='selUser' style="width: 25.3em;">
<option value="Varotto_Lab" name="Varotto_Lab">Varotto Lab</option>
<option value="Scanlon_Lab" name="Scanlon_Lab">Scanlon Lab</option>
<option value="Pereira_Lab" name="Pereira_Lab">Pereira Lab</option>
<option value="Kaeppler_and_Walley" name="Kaeppler_and_Walley">Kaeppler and Briggs Lab</option>
<option value="Springer_Lab" name="Springer_Lab">Springer Lab</option>
<option value="Hochholdinger_Lab" name="Hochholdinger_Laby">Hochholdinger Lab</option>
<option value="Fowler_Lab" name="Fowler_Lab">Fowler Lab</option>
<option value="avg_gene_abundance" name="avg_gene_abundance">Average gene abundance</option>
<option value="gene_breadth" name="gene_breadth">Gene breadth</option>
<option value="Max_FPKM" name="Max_FPKM">Max FPKM</option>
</optgroup>
</select>
    <h4>Labels</h4>
    <div class="bs-multiselect">
<select id = "mltislct2" name="comp_select" multiple = "multiple">
<option value="no_label" name="no_label">No Label</option>
<option value="classical_label" name="classical">Classical</option>
<option value="core_label" name="core">Pan-genome</option>
<option value="Origin" name="Origin">Gene Origin</option>
</select>
</div>
</div>
<br>
<br>
<div style="width:200px;height:85px;margin-left:22px" class="btn1">
<button style="background-color: #fff; /* Green */
  border: none;
  color: #7386D5;
  padding: 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 20px;
  margin: 1px 1px;
  cursor: pointer;
  width: 14em;
border-radius: 12px;" class="button button4" onclick="submitForm('/GeneExp)" id="bttn1">Submit for table</button>
</div>
    <div class="container">
        <label for="cluster" style="font-size: 18px;font-family: inherit;font-weight: 500;
    line-height: 1.1;
    color: inherit; cursor: help;"  title="Cluster is only used for the Downsampled Hierarchical scatter plots analysis">Choose cluster:</label>
<input type="number" id="cluster" name="cluster" style="color:black; width: 100px;  border-radius: 4px;
box-shadow: none;border-color:white;" title="Cluster is only used for the Downsampled Hierarchical scatter plots analysis" min="0" max="10"><br><br>
<h4>Choose analysis</h4>
<select name="comp_analysis" style=" width: 25.3em;"  id='selUser1' class="form-control" >
    <optgroup label="Graphical analysis">
    <option value="none" style="color:black">Select one analysis</option>
    <option value="histogram">Histogram</option>
    <option value="Count_and_distribution">Count and distribution</option>
    <option value="Pair_plots">Pair plots</option>
    <option value="Box_plots">Box plots</option>
    <option value="Violin_plots">Violin plots</option>
    <option value="Joint_plots">Joint plots</option>
    <option value="Scatter_plots">Scatter plots</option>
    <option value="Correlation_plots">Correlation plots</option>
        <option value="Count_downsampled_dendogram_plot">Downsampled Dendogram plot</option>
    <option value="Count_downsampled_hierarchial_scatter_plot">Downsampled Hierarchial Scatterplot</option>
        <option value="Count_downsampled_Hierarchial_heatmap">Downsampled Hierarchial Heatmap</option>
    <option value="Heatmap_plots">Heatmap</option>
    <option value="PCA_2D_samples_plots">PCA 2D variables cluster</option>
        <option value="PCA_2D_label_plots">PCA 2D observation cluster</option>
        <option value="PCA_2D_biplot_plots">PCA 2D biplot cluster</option>
    <option value="PCA_3D_samples_plots">PCA 3D variables cluster</option>
    <option value="PCA_3D_label_plots">PCA 3D observation cluster</option>

        <!--<option value="PCA_3D_biplot_plots">PCA 3D biplot cluster</option>-->
    </optgroup>
</select></div>

    <br>
<br>

<div style="width:200px;height:85px;margin-left:22px" class="btn2">
<button style="background-color: #fff; /* Green */
  border: none;
  color: #7386D5;
  padding: 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 20px;
  margin: 1px 1px;
  cursor: pointer;
  width: 14em;
border-radius: 12px;" class="button button5" onclick="submitForm('/GeneExp_plot')"  id="bttn2">Submit for analysis</button>
</div>

<div style="width:200px;height:85px;margin-left:22px" class="btn3">
<button style="background-color: #fff; /* Green */
  border: none;
  color: #7386D5;
  padding: 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 20px;
  margin: 1px 1px;
  cursor: pointer;
  width: 14em;
border-radius: 12px;" class="button button6" onclick="submitForm('/downsampled_GeneExp_plot')"  id="bttn3">Downsampled analysis</button>
</div>

</form>
                <ul class="list-unstyled CTAs">
                    <li><a href="static/collection_zip/gene_exp_features.zip" class="download">Download source</a></li>
                    <li><a href="static/collection_zip/All_data.zip" class="article">Download all</a></li>
                </ul>
            </nav>
     <!-- Page Content Holder -->
            <div id="content">

                <nav class="navbar navbar-default">
                    <div class="container-fluid">

                        <div class="navbar-header">
                            <button type="button" id="sidebarCollapse" class="navbar-btn">
                                <span></span>
                                <span></span>
                                <span></span>
                            </button>
                        </div>

                        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                            <ul class="nav navbar-nav navbar-right">
                                 <li><a class="navbar-brand" href="/">Home</a></li>
                                <li><a class="navbar-brand" href="/features">Features and Analysis</a></li>
                                <!--<li><a class="navbar-brand" href="#">Models</a></li>-->
                            </ul>
                        </div>
                    </div>
                </nav>

                <p style="color:blue;font-size:20px;">Please go through the below explanation for details on table and plot
                    interpretations</p>
                <h3>Table column interpretation:</h3>

               <b> Within Gene Expression input box, we have the lab names from where gene expression data have been collected
                    for different tissues. Therefore to reduce complexity and time, both for the table and for the choose analysis options:
                   Histogram, Count and distribution plot, Pair plots, Box plots, Violin plots, Joint plots, Scatter plots,
                   Correlations plots and Downsampled Hirarchial Scatter plot only the first five tissues from the selected lab is
                   displayed wheareas for all other plots (PCA [2D,3D] , Heatmap and Downsampled Hieararhial Heatmap)
               all tissues from the selected lab were considered. </b>
                <br>
                <br>
                <p>The following papers [by lab] were the sources of the RNA-seq datasets included in this interface:<p></p>
                <h4>1. Varotto Lab</h4>
                <p><a href= "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4962059/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4962059/
                </a></p>
                <h4>2. Scanlon Lab</h4>
                <p><a href= "https://academic.oup.com/plcell/article/26/12/4718/6102403">https://academic.oup.com/plcell/article/26/12/4718/6102403
                </a></p>
                <h4>3.  Pereira Lab  </h4>
                <p><a href= "https://academic.oup.com/plphys/article/160/2/846/6109548">https://academic.oup.com/plphys/article/160/2/846/6109548
                </a></p>
                <h4>4. Kaeppler and Briggs Lab </h4>
                <p>Maize Atlas Stelpflug 2015 [Kaeppler Lab]: &nbsp
                    <a href= "https://pubmed.ncbi.nlm.nih.gov/27898762/">https://pubmed.ncbi.nlm.nih.gov/27898762/
                </a>
                    <br>
                    Walley 2016 [Briggs Lab] : &nbsp
                    <a href= "https://www.science.org/doi/full/10.1126/science.aag1125">https://www.science.org/doi/full/10.1126/science.aag1125
                </a>
                </p>
                <h4>5. Springer Lab </h4>
                 <p><a href= "https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004915">https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004915
                </a></p>
                <h4>6. Hochholdinger Lab </h4>
                 <p><a href= "https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-741">https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-741
                </a></p>
                <h4>7. Fowler Lab </h4>
                <p><a href= "https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1008462">https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1008462
                </a></p>
                <h4>8. Average gene abundance </h4>
                <p> This is average gene expression level of each gene across all tissues collected from different labs mentioned above.
                </p>
                <h4>9. Gene breadth </h4>
                <p> It is the number of tissues in which a gene is expressed.
                </p>
                <h4>10. Max FPKM </h4>
                <p>
                    This is maximum gene expression level of each gene across all tissues collected from different labs mentioned above.
                </p>


                <br>
                <h4>10. Labels </h4>
                <p>The labels are the classes or the groups the genes are mapped into.The labels can act as both target variable or feature
                as per the need of the user for solving their specific problem </p>
                <h4>10.1 No Label</h4>
                <p>This selection is provided to enable users to view the properties of all genes without labeling
                    them into different gene categories or annotations. This is to let users examine the features of
                    multiple genes and identify common patterns among them. As it involves the inspection of
                all the genes therefore they work only for "Submit for analysis" button .</p>
                <h4>10.2 Classical Genes</h4>
                <p>Classical genes can be defined as the most well-studied genes mainly for their visible mutant phenotype (for example: liguleless3). </p>
                <h4>10.3 Pan-genome Genes </h4>
                <p>A gene in a given taxonomic group is either present in every individual (core), or absent in at least a single individual (dispensable). </p>
                <h4>10.4 Origin Genes </h4>
                <p>Gene duplication is an important evolutionary mechanism allowing new genetic material and thus opportunities to acquire new gene functions for an organism. There are different origins of duplications such as whole-genome duplications, tandems, etc.  </p>
                <br>
               <h3>Graph interpretations:</h3>

                <p>To the top right corner of the plots/graphs, there are options to download plot, zoom-out/zoom in, reset axes, autoscale, toggle spike lines, show closest data on hover, compare data on hover, box select, pan and lasso. Users can also select specific legends to view data only for the selected legends. Details on the interactive plot options are available here:                 <br>
                    <a href = https://plotly.com/chart-studio-help/getting-to-know-the-plotly-modebar/>Interactive graph features</a>

                </p>


                <h4>1. Histogram</h4>
                <p>The Histogram shows the frequency distribution of the selected lab tissue such as in Varotto lab
                    tissues: rmr6_leaf_control_T0, rmr6_leaf_control_T7, rmr6_leaf_drought_salt_T0 etc.
                The X-axis in the histogram represents the range of values present in the selected lab tissue.
                The Y-axis represents the frequency of values. In addition to the graph,to increase the interpretability of the data,
                We have also included P-values,mean and standard deviations of the selected datasets. </p>

                <h4>2. Count and distribution</h4>
                <p>The Count and distribution plot is a smoothed, continuous version of a histogram estimated from
                    the data. The most common form of estimation is known as kernel density estimation.The x-axis is
                    the value of the selected lab tissue just like in a histogram and the y-axis in a distribution plot
                    is the probability density function and not a probability. The difference is the probability density
                    is the probability per unit on the x-axis. In general the y-axis on the distribution plot is a value
                    only for relative comparisons between different categories like classical and other genes.</p>
                <h4>3. Pair plots</h4>
                <p>The pairs plot builds on two basic figures, the distributions and the scatter plot. The distributions on the
                    diagonal allows us to see the distribution of a single selected lab tissue while the scatter plots
                    on the upper and lower triangles show the relationship (or lack thereof) between two variables such as
                    rmr6_leaf_control_T0 and  rmr6_leaf_control_T7. </p>
                <h4>3. Box plots</h4>
                <p>Boxplots are a measure of how well distributed the data in the selected lab tissue's are. It divides the data set
                    into three quartiles. This graph represents the minimum, maximum, median, first quartile and third quartile in the
                    selected lab tissue. It is also useful in comparing the distribution of data across data sets by drawing
                    boxplots for each of them such as the core, non-core, dispensable and private genes. Boxplots can be used to:<br>
                    <ul>
                      <li>Identify outliers or anomalous data points</li>
                      <li>To determine if our data is skewed</li>
                      <li>To understand the spread/range of the data</li>
                    </ul>
                </p>
                <h4>4. Violin plots</h4>
                <p>Violin plots are similar to box plots, except that they also show the probability density of the selected lab tissue
                    at different values. These plots include a marker for the median of the data and a box indicating the
                    interquartile range, as in the standard box plots. Overlaid on this box plot is a kernel density estimation.
                    Like box plots, violin plots are used to represent comparison of a variable distribution such as
                    rmr6_leaf_control_T0, rmr6_leaf_control_T7
                    (or sample distribution)  across different "categories" (Classical/Other or Core/Non-core) .
                    <br>A violin plot is more informative than a plain box plot. In fact while a box plot only shows summary statistics
                    such as mean/median and interquartile ranges, the violin plot shows the full distribution of the selected
                    lab tissue.</p>
                <h4>5. Joint plots</h4>
                <p>
                    A Jointplot comprises three plots. Out of the three, one plot displays a bivariate graph(scatter plot) which shows how the
                    one variable( such as rmr6_leaf_control_T0) varies with the another variable(such as Exon number). Another plot is
                    placed horizontally at the top of the bivariate graph and it shows the distribution in the form of
                    marks along an axis for each one of the selected lab tissue's ( for example rug plot of rmr6_leaf_control_T0 and rmr6_leaf_control_T7).
                    The third plot is placed  on the right margin of the bivariate graph with the orientation set to vertical and
                    it shows the distribution of again the two selected lab tissue's .
                    <br>It is very helpful to have univariate and bivariate plots together in one figure.
                    This is because the univariate analysis focuses on one variable, it describes, summarizes and shows any
                    patterns in your data and the bivariate analysis explores the relationship between two variables and also
                    describes the strength of their relationship.</p>
                <h4>6. Scatter plots</h4>
                <p>A Scatter plot is a great way of exploring relationships or patterns in data. But adding a regression line can
                    make those patterns stand out . Therefore Scatter plot with simple linear regression for the selected lab tissue's,
                    explains the strength of the relationship between the two variables such as rmr6_leaf_control_T0 or rmr6_leaf_control_T7 in your scatter-plot using R2,
                    the squared correlation coefficient.It is always between 0 and 1. Higher R2 indicates stronger relationship between
                the selected lab tissue's . </p>
                <h4>7. Correlation plots</h4>
                <p> Correlation heatmap is graphical representation of correlation matrix representing correlation between different
                    selected lab tissue's. Correlation ranges from -1 to +1. Values closer to zero means there is no linear
                    trend between the two variables. The closer to 1 the correlation is the more positively correlated they are;
                    that is as one increases so does the other and the closer to 1 the stronger this relationship is. Correlation
                plots also alerts us to potential multicollinearity problems.</p>
                <h4>8. Downsampled Dendrogram plots</h4>
                <p>A dendrogram is a type of tree diagram showing hierarchical clustering — relationships between similar sets of
                    genes based on the selected lab tissue's. They are frequently used in biology to show clustering between genes
                    or samples, but they can represent any type of grouped data.The dendrogram is built on downsampled data to save
                time and complexity. <br>
                The branches in the dendrogram are arranged according to how similar (or dissimilar) they are. branches that are
                    close to the same height are similar to each other; branches with different heights are dissimilar —
                    the greater the difference in height, the more dissimilarity. Also the different clusters of genes based on the
                    selected lab tissue's are marked by different colors.</p>
                <h4>9. Downsampled Hierarchical Scatterplot</h4>
                <p>The Hierarchical scatter plot is a type of pair plot that can be used to visualize the relationship of different
                    pairs of lab tissue's on the clusters of genes formed during dendrogram.Therefore to create Hierarchial
                    scatterplot we need to input the number of clusters in the choose cluster input box. For
                    the Downsampled Hierarchical Scatter plot, we need to specify the number of clusters we
                    want to view. Moreover, this number of clusters corresponds to the number of clusters
                    formed in the Dendrogram plot, so we perform this analysis only after studying the
                    Dendrogram plot. The Downsampled Hierarchical Scatter plot is dynamically sized for the
                    number of clusters, so every user can analyze the dataset based on how many clusters
                    they believe the selected data is forming .
                </p>
                <h4>10. Downsampled Hierarchical Heatmap</h4>
                <p>Hierarchical clustering heatmap is an intuitive way to visualize information from complex data. It’s also called
                    a false colored image, where data values are transformed to color scale. Heat maps allow us to simultaneously
                    visualize clusters of samples(selected lab tissue's) in the column and features (genes) in the rows. <br>First
                    hierarchical clustering is done of both the rows and the columns of the data matrix. The columns/rows of the
                    data matrix are re-ordered according to the hierarchical clustering result, putting similar observations close
                    to each other. The blocks of ‘high’ and ‘low’ values are adjacent in the data matrix. Finally,
                    a color scheme is applied for the visualization and the data matrix is displayed. Visualizing the data
                    matrix in this way can help to find the variables(selected lab tissue's) that appear to be characteristic
                    for each sample cluster.</p>

                <h4>10. Heatmap</h4>
                <p>A heatmap is a plot of rectangular data as a color-encoded matrix.This is a great way to visualize data, because
                    it can show the relation between variables (selected lab tissue's) including genes. The heatmap here shows
                the relation between the first 100 genes and the selected lab tissue's values to reduce time and complexity .</p>
                <h4>PCA</h4>
                <P>PCA reduces the high-dimensional interrelated data to low-dimension by linearly transforming the old variable
                    into a new set of uncorrelated variables called principal component (PC) while retaining the most possible
                    variation. <br>
                    The first component has the largest variance followed by the second component and so on.
                    The first few components retain most of the variation, which is easy to visualize and summarize the features
                    of original high-dimensional datasets in low-dimensional space. PCA helps to assess which original samples are
                    similar and different from each other. <br>
                    In our case when the selected number of lab tissue's are higher than 2 then, it is arduous to visualize them at
                    the same time to interpret the genes. PCA transforms them into a new set of variables (PCs) with top PCs
                    having the highest variation. PCs are ordered which means that the first few PCs (generally first 3 PCs but
                    can be more) contribute most of the variance present in the original high-dimensional dataset. These top
                    first 2 or 3 PCs can be plotted easily and summarize the features of all original variables (selected lab tissue's).
</P>
                <h4>11. PCA 2D variables cluster</h4>
                 <p>
                    A PCA 2D variables cluster  plot shows how strongly each characteristic or variable
                    influences a principal component. In the plot we can see these vectors(variables)
                    are pinned at the origin of PCs (PC1 = 0 and PC2 = 0). Their project values on each PC show
                    how much weight they have on that PC.
                     <br>
                     <br>
                     The data points are annotated by grouped tissue or treatment name . For example: PCA
                     analysis output with the "Scanlon lab" displays preblade_adaxial tissue annotation in dark blue
                     which comprises of three replicates of the tissue : B L1 1 preblade
                     adaxial, B L1 2 preblade adaxial and B L1 3 preblade adaxial. The grouped annotation
                     is done to make it easier to identify if same tissue from a lab treated
                     under different time points cluster together or not.
                     From the plot, we observe that two preblade adaxial are placed together
                     whereas one of them is positioned slightly further.
                </p>
                <h4>12. PCA 2D observation cluster</h4>
                <p>A PCA 2D observation cluster plot showing clusters of samples/observations based on their similarity.PCA does not discard
                    any samples or observations. Instead, it reduces the overwhelming number of dimensions by
                    constructing principal components (PCs). PCs describe variation and account for the varied influences
                    of the original characteristics. Such influences, or loadings, can be traced back from the PCA plot
                    to find out what produces the differences among clusters. <br>
                    Observations further out are either outliers or naturally extreme observations. Plot observations are annotated with shapes
                    and colors to highlight gene models and gene types for example (Classical/Others). In this way, we can easily hover
                    over any data point, especially outliers, to find gene models with extreme observation values across various sample types.
                    (selected lab tissue's)

                </p>
                <h4>13. PCA 2D biplot cluster</h4>
               <p> PCA biplot = PCA observations plot + variable plot. <br>
                    we can probably notice that a PCA biplot simply merges an usual observations PCA plot with a plot of
                    variables. The arrangement is like this: <br>

                    <b>*</b>Bottom axis: PC1 score. <br>
                    <b>*</b> Left axis: PC2 score. <br>
                    In other words, the left and bottom axes are of the PCA observations plot — use them to read PCA scores of the
                    observations (dots).
                    </p>
                <p>
                    Another nice thing about these plots: the angles between the vectors/variables tell us how
                    characteristics or variables correlate with one another. <br>
                    <b>*</b> When two vectors are close, forming a small angle, the two variables they represent are positively
                    correlated. <br>
                    <b>*</b>If they meet each other at 90°, they are not likely to be correlated. <br>
                    <b>*</b>When they diverge and form a large angle (close to 180°), they are negatively correlated.
                </p>

                <h4>14. PCA 3D variables cluster</h4>
                <p> This is similar to the PCA 2D variables cluster plot but now instead of two  principal components, we can have three PC'S that contribute to
                    most of the variance present in the original high-dimensional dataset. These top
                    3 PCs can be plotted easily in 3 dimensional space and summarize the features of all original variables
                    (selected lab tissue's) .</p>
                <h4>15. PCA 3D observation cluster</h4>
                <p> This is similar to the PCA 2D observation cluster but now instead of two  principal components, we can have three PC'S that contribute to
                    most of the variance present in the original high-dimensional dataset. These top
                    3 PCs can be plotted easily in 3 dimensional space and summarize the variance in the observations . <br>
                 A PCA 3D observation cluster plot displays how much variation each principal component captures from
                    the data. If the first three PCs are sufficient to describe the essence of the data.</p>
                <b>For further details on these features please go through the below tool :</b>
                <br>

                <a href= " https://qteller.maizegdb.org/">https://qteller.maizegdb.org/</a>



            </div>
    </div>

<script>
var dataSet = {{ select|safe }}

var my_columns = [];

$.each( dataSet[0], function( key, value ) {
        var my_item = {};
        my_item.data = key;
        my_item.title = key;
        my_columns.push(my_item);
});

$(document).ready(function() {
  $('#example').DataTable({
    "scrollX": true,
    data: dataSet,
    "columns": my_columns
  });
});
</script>

<script>
$(document).ready(function() {
$('#mltislct2').multiselect({
multiple: false,
maxHeight: 200,
buttonWidth:'350px',
enableFiltering: true,
enableCaseInsensitiveFiltering: true,
includeSelectAllOption: false,
filterPlaceholder:'Search Here..'

});
});
</script>
<script type="text/javascript">
 $(document).ready(function () {
     $('#sidebarCollapse').on('click', function () {
         $('#sidebar').toggleClass('active');
         $(this).toggleClass('active');
     });
 });
</script>
<script>
 function submitForm(action) {
    var form = document.getElementById('framework_form');
    form.action = action;
    form.submit();
  }
</script>

<script>
    $(document).ready(function(){

  // Initialize select2
  $("#selUser").select2();
});
</script>
<script>
    $(document).ready(function(){

  // Initialize select2
  $("#selUser1").select2();

});
</script>
</body>
</html>